EventSourcing.Cosmos/CosmosEventStore.cs

```diff
@@ -80,26 +97,45 @@ public async Task AddAsync(IList<TBaseEvent> events, CancellationToken cancellat
     {
       if (events == null) throw new ArgumentNullException(nameof(events));
       if (events.Count == 0) return;
-      
+
       await VerifyAsync(events);
-      
-      var batch = _container.CreateTransactionalBatch(new PartitionKey(events.First().AggregateId.ToString()));
+
+      var batch = _eventsContainer.CreateTransactionalBatch(new PartitionKey(events.First().AggregateId.ToString()));
       foreach (var @event in events) batch.CreateItem(@event, _batchItemRequestOptions);
       var response = await batch.ExecuteAsync(cancellationToken);
 
       if (!response.IsSuccessStatusCode) await ThrowAsync(response, events);
     }
     
-    private async Task<bool> ExistsAsync(Guid aggregateId, uint version)
+    /// <summary>
+    /// AddSnapshotAsync: Store snapshot as a <see cref="TBaseEvent"/>s to the Cosmos Event Store
+    /// </summary>
+    /// <param name="snapshot"><see cref="TBaseEvent"/>s to add</param>
+    /// <param name="cancellationToken">Cancellation Token</param>
+    /// <exception cref="InvalidOperationException">Thrown when snapshot container is not provided</exception>
+    /// <exception cref="ArgumentException">Thrown when trying to add <see cref="TBaseEvent"/>s with empty AggregateId</exception>
+    /// <exception cref="EventStoreException">Thrown when conflicts occur when storing <see cref="TBaseEvent"/>s</exception>
+    /// <exception cref="ConcurrencyException">Thrown when storing <see cref="TBaseEvent"/>s</exception> with existing partition key and version combination
+    public async Task AddSnapshotAsync(TBaseEvent snapshot, CancellationToken cancellationToken = default)
```

Comment:
Adding the ```AddSnapshotAsync``` method scatters a lot of ```if container exists``` logic everywhere. I would opt to maybe subclass the EventStore, which requires a snapshot container as a parameter, and adding the method there, such that these checks don't have to be done at runtime. It would benefit both readability (less checks everywhere) and robustness (if a user didn't provide the correct options, they will figure out on startup, rather than the first time they try to persist or rehydrate a snapshot)